src/rust/storage/seg/src/store.rs

```diff
@@ -0,0 +1,14 @@
+/// Copies `size` bytes at `byte_ptr` to the `offset` of `data`
+/// Returns the next `offset`, that is, the next byte of `data` to be copied into
+pub fn store_bytes_and_update_offset(byte_ptr: *const u8, offset: usize, size: usize, data: &mut [u8]) -> usize {
+    // get corresponding bytes from byte pointer
```

Comment:
I put this function here so that `Segments`, `TtlBuckets` and `HashTable` could all access it. I am slightly concerned about security as I am not sure if this could be used as a "write what I want where I want"